connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReaderAdmin.scala

```diff
@@ -504,7 +512,9 @@ private[kafka010] class KafkaOffsetReaderAdmin(
       if (untilOffset < fromOffset) {
         reportDataLoss(
           s"Partition $tp's offset was changed from " +
-            s"$fromOffset to $untilOffset, some data may have been missed",
+            s"$fromOffset to $untilOffset. This could either be a user error that the start " +
```

Comment:
I'd propose to rephrase the 2nd sentence a bit like below:

> This could be either 1) a user error that the start offset is set beyond available offset when starting query, or 2) the topic partition may have deleted and re-created.

Please check whether the above sentence misses some case you want to describe or you find a wrong information. We can just modify a bit to make it be correct.